packages/Webkul/Checkout/src/Cart.php

```diff
@@ -446,6 +446,19 @@ public function saveShippingMethod($shippingMethodCode): bool
             return false;
         }
 
+        $shippingMethods = Shipping::collectRates()['shippingMethods'];
+        $isMethodAvailable = false;
+        foreach ($shippingMethods as $shippingMethod) {
+            foreach ($shippingMethod['rates'] as $rate)
+            if ($rate->method == $shippingMethodCode) {
+                $isMethodAvailable = true;
+                break 2;
+            }
+        }
+        if (!$isMethodAvailable) {
+            return false;
+        }
+
```

Comment:
This can be more shorten, if you check this method also,
https://github.com/bagisto/bagisto/blob/a832c34644434a5149f1ad9df8e48ade8e9b999b/packages/Webkul/Shipping/src/Shipping.php#L129

Reply:
Hi @devansh-webkul
There's a point I can't understand so for example, we want flatrate to be set on cart, the code that sends to method is flatrate_flatrate (shipping method) but in the code you suggest it uses shipping class string that is saved in the config and creates a class of that so for my example it will be 'flatrate'.
but I don't have flatrate in here so that I can't use something like Config::get("carriers.{$shippingMethodCode}") and then get class of that 
so in summary my question is how to get a shipping class with shipping method code like flatrate_flatrate?
we can use other approaches like adding the shipping method code to the keys of the array where returns in getShippingMethod facade  and just check that if the key exists

Reply:
Hi @AmooAti,

You can create one more protected property named as `method` which will get access same as `code`.

Then you can update this method i.e. `getShippingMethods` and update the key like,

~~~php
'code' => $this->getCode(),
'method' => $this->getMethod(),
~~~

Now the existing method will be updated, now you can check with `in_array`.